x-pack/plugin/esql/qa/security/src/javaRestTest/java/org/elasticsearch/xpack/esql/EsqlSecurityIT.java

```diff
@@ -674,6 +685,20 @@ public void testLookupJoinFieldLevelSecurity() throws Exception {
             )
 
         );
+
+        resp = runESQLCommand("fls_user4", "ROW x = 40.0 | EVAL value = x | LOOKUP JOIN `lookup-user2` ON value");
+        assertOK(resp);
+        respMap = entityAsMap(resp);
+        assertThat(
+            respMap.get("columns"),
+            equalTo(
+                List.of(
+                    Map.of("name", "x", "type", "double"),
+                    Map.of("name", "value", "type", "double"),
+                    Map.of("name", "org", "type", "keyword")
+                )
+            )
+        );
     }
```

Comment:
Sorry, one last test suggestion - we can also add this in a separate PR if you're already about to make this ready to merge.

I just noticed: all of the fls test cases have permission to read the `value` field, which is the join key.

I think it'd be interesting if forbidding this field for a user makes it so all look ups are null. And, if we wanna be extra paranoid, consider the case where the lookup index has duplicates for a given join key, so the `LOOKUP JOIN` would lead to extra rows - but without permissions granted on the join key, no extra rows should be emitted. (Otherwise, users could exfiltrate statistics about fields they have no permissions on).

Reply:
Thanks Alex, I still have a couple of things to push, I'll add this one as well.
I _think_ the whole query will fail because the user doesn't see the join key in the schema, but I'll double-check

Reply:
The new tests confirm the above: if a role cannot see the join key in the lookup index, the query will fail with "Unknown column"